public/app/plugins/datasource/influxdb/query_ctrl.ts

```diff
@@ -275,14 +288,40 @@ export class InfluxQueryCtrl extends QueryCtrl {
     this.target.rawQuery = !this.target.rawQuery;
   }
 
-  getMeasurements(measurementFilter: any) {
+  getMeasurementsUnlimited(measurementFilter: any) {
     const query = this.queryBuilder.buildExploreQuery('MEASUREMENTS', undefined, measurementFilter);
+
     return this.datasource
       .metricFindQuery(query)
       .then(this.transformToSegments(true))
       .catch(this.handleQueryError.bind(this));
   }
 
+  // our measurement-throttling approach will cause some
+  // of the getMeasurement-promises to stay unresolved.
+  // with the following timeout-code we make sure their
+  // handling wil be finished either way, meaning, every getMeasurements
+  // call coming from the "outside" will either resolve with data,
+  // or (after some time) be rejected. i am somewhat worried
+  // that unresolved promises might hold references to resources
+  // and cause memory problems eventually, so we do this to be sure.
+  getMeasurementsUnlimitedWithTimeout(measurementFilter: any) {
```

Comment:
this is the mentioned reject-after-long-waiting code. i'm not sure if this is needed or not.